userguide/storage.rst

```diff
@@ -713,53 +713,70 @@ existing datasets to close them--the remaining
-data is written and stored. However, deduplication is RAM intensive. A
-general rule of thumb is 5 GB of RAM per terabyte of deduplicated
-storage. **In most cases, compression provides storage gains
-comparable to deduplication with less impact on performance.**
-
-In %brand%, deduplication can be enabled during dataset creation. Be
-forewarned that **there is no way to undedup the data within a dataset
-once deduplication is enabled**, as disabling deduplication has
-**NO EFFECT** on existing data. The more data written to a
-deduplicated dataset, the more RAM it requires. When the system starts
-storing the DDTs (dedup tables) on disk because they no longer fit
-into RAM, performance craters. Further, importing an unclean pool can
-require between 3-5 GB of RAM per terabyte of deduped data, and if the
-system does not have the needed RAM, it will panic. The only solution
-is to add more RAM or recreate the pool.
-**Think carefully before enabling dedup!**
-This `article
-<http://constantin.glez.de/blog/2011/07/zfs-dedupe-or-not-dedupe>`_
-provides a good description of the value versus cost considerations
-for deduplication.
-
-**Unless a lot of RAM and a lot of duplicate data is available, do not
-change the default deduplication setting of "Off".**
-For performance reasons, consider using compression rather than
-turning this option on.
-
-If deduplication is changed to *On*, duplicate data blocks are removed
-synchronously. The result is that only unique data is stored and
-common components are shared among files. If deduplication is changed
-to *Verify*, ZFS will do a byte-to-byte comparison when two blocks
-have the same signature to make sure that the block contents are
-identical. Since hash collisions are extremely rare, *Verify* is
-usually not worth the performance hit.
-
-.. note:: After deduplication is enabled, the only way to disable it
-   is to use the :samp:`zfs set dedup=off {dataset_name}` command
-   from :ref:`Shell`. However, any data that has already been
-   deduplicated will not be un-deduplicated. Only newly stored data
-   after the property change will not be deduplicated. The only way to
-   remove existing deduplicated data is to copy all of the data off of
-   the dataset, set the property to off, then copy the data back in
-   again. Alternately, create a new dataset with
-   :guilabel:`ZFS Deduplication` left disabled, copy the data to the
-   new dataset, and destroy the original dataset.
+#ifdef freenas
+.. index:: Deduplication
+.. _Deduplication:
+
+Deduplication
+^^^^^^^^^^^^^
+
+Deduplication ("dedup") is a process where ZFS aims to store a single copy
+of any data that is repeated in different blocks or files, rather than storing 
+identical copies each time the data occurs. Deduplication 
+potentially allows repeated and identical data to be stored very 
+efficiently in a smaller space, but also places a 
+very heavy burden on RAM and on the CPU. In particular, dedup 
```

Comment:
Rework the sentence to get rid of 'potentially'. That makes it sound like we aren't sure it works. Should stick to confirmed information. Get rid of all ambiguous "data" terms. Refer to block data specifically. 
Remove "in particular" and "dedup" start with the full word Deduplication. We aren't paying by the character ;) Rework the sentence to describe how the data is stored. This should hopefully allow the removal of ", but" split into a separate sentence if need be.

Reply:
Again, for a user storing files, what they see is the file being deduped, they **never** see "blocks" being deduped, nor do we need to dive into files being stored in blocks and blocks being the actual basis of deduplication. It just doesn't add anything, pragmatically, if the user knows this subtlety having not known it before.  (See previous comment). It's like insisting a user knows about unix pipes or 512B/4K disk sectors when discussing file backup, No actual user benefit other than engineering satisfaction :)

Did the rest.

Reply:
Again, for a user storing files, what they see is the file being deduped, they **never** see "blocks" being deduped, nor do we need to dive into files being stored in blocks and blocks being the actual basis of deduplication. It just doesn't add anything, pragmatically, if the user knows this subtlety having not known it before.  (See previous comment). It's like insisting a user knows about unix pipes or 512B/4K disk sectors when discussing file backup, No actual user benefit other than engineering satisfaction :)

Did the rest.